src/core/thread/thread_netif.cpp

```diff
@@ -171,6 +172,23 @@ ThreadError ThreadNetif::RouteLookup(const Ip6::Address &source, const Ip6::Addr
     return error;
 }
 
+ThreadError ThreadNetif::TmfFilter(const otMessage *aMessage, const otMessageInfo *aMessageInfo)
+{
+    const Ip6::MessageInfo &messageInfo = *static_cast<const Ip6::MessageInfo *>(aMessageInfo);
+    ThreadError error = kThreadError_None;
+
+    VerifyOrExit(messageInfo.GetSockAddr().IsRoutingLocator() ||
+                 messageInfo.GetPeerAddr().IsRoutingLocator() ||
+                 messageInfo.GetSockAddr().IsLinkLocal() ||
+                 messageInfo.GetSockAddr().IsAnycastRoutingLocator() ||
+                 messageInfo.GetPeerAddr().IsAnycastRoutingLocator(),
```

Comment:
Reading the spec again, you are correct.  At the same time, I think the spec should be updated to be more strict and say both of the following rules must match:

- Source Address is RLOC, ALOC, or link-local address.
- Destination Address is RLOC, ALOC, or link-local address.

In other words, we should not allow messages to pass that have global IPv6 source address with RLOC destination address, for example.